evm/ts/src/ntt.ts

```diff
@@ -143,19 +141,30 @@ export class EvmNtt<N extends Network, C extends EvmChains>
     return enabled.filter((x) => x).length > 0;
   }
 
-  getIsExecuted(attestation: Ntt.Attestation): Promise<boolean> {
-    const { emitterChain: chain, payload } =
-      attestation as VAA<"Ntt:WormholeTransfer">;
-    return this.manager.isMessageExecuted(
-      Ntt.messageDigest(chain, payload["nttManagerPayload"])
+  async getIsExecuted(attestation: Ntt.Attestation): Promise<boolean> {
+    const payload =
+      attestation.payloadName === "WormholeTransfer"
+        ? attestation.payload
+        : attestation.payload.payload;
+    const isExecuted = await this.manager.isMessageExecuted(
+      Ntt.messageDigest(attestation.emitterChain, payload["nttManagerPayload"])
     );
+    if (!isExecuted) return false;
+    // Also check that the transfer is not queued for it to be considered complete
+    const queued = await this.getInboundQueuedTransfer(
+      attestation.emitterChain,
+      payload["nttManagerPayload"]
+    );
```

Comment:
Should we have a separate `isQueued` fn on the interface?

Reply:
Maybe it would make the code cleaner, I can add that